spyder/plugins/preferences/api.py

```diff
@@ -809,7 +817,11 @@ def prompt_restart_required(self):
         answer = QMessageBox.information(self, msg_title, msg,
                                          QMessageBox.Yes | QMessageBox.No)
         if answer == QMessageBox.Yes:
-            self.main.application.sig_restart_requested.emit()
+            self.restart()
+
+    def restart(self):
+        """Restart Spyder."""
+        self.main.restart(close_immediately=True)

```

Comment:
After PR https://github.com/spyder-ide/spyder/pull/15629, we now emit `sig_restart_requested` from the Application plugin, so that other plugins can connect to it and perform actions when a restart happens.

So, you need to find a way of using that signal, as it was before.